tracing/tracing.go

```diff
@@ -2,16 +2,16 @@
 package tracing
 
 import (
-	"context"
 	"net/http"
 
 	"github.com/birdie-ai/golibs/slog"
 	"github.com/google/uuid"
 )
 
-// Instrument will instrument the given handler by adding tracing context on the
-// request context.
-func Instrument(h http.Handler) http.Handler {
+// InstrumentHTTP will instrument the given [http.handler] by adding a slog.Logger on the request context.
+// The logger will have `trace_id` added to it.
+// Use slog.FromCtx(ctx) to retrieve the logger.
+func InstrumentHTTP(h http.Handler) http.Handler {
```

Comment:
Also, not really happy about this either, I'm wondering if this is tracing, or this is HTTP middleware (short answer will be, both 😂)

What do you think of starting a `http` package, this would be the middleware `http.Instrument(h)`, I think about the Org-ID for example, the implementation is exactly the same, would we call organization package? or `http.Auth()` maybe?

I think we'll at some point have a tracing package, but I feel here is not really tracing is just reading a header (like org-id will), what do you think?

Reply:
I think I was thinking something really different, but thinking better, maybe it's good the way that you did.

I was expecting to have all HTTP related stuff together, but as we're splitting different libs, then HTTP lib would define stuff that are tracing related (like the field name and header), which would be weird, if they were all in the same library I think would be ok.

Let's move on, like that, I think can be good!

another minor comment, is if Instrument is the best wording, we can move forward as I don't really have other options in my mind.

Reply:
Im very torn on these different design decisions, not strong feelings so far. If we stick with tracing I think tracing.InstrumentHTTP could be enough since it should have only a single tracing related function (but I specified the procotol because we could have other forms of instrumentation that are tracing related but are not HTTP...maybe actual tracing =P).

Regarding the OrgID, I would also place it on the tracing.InstrumentHTTP..I just want something that solves all our problems regarding basic traced data on HTTP..but maybe that is bad...and in that case we would need a different package like organization.InstrumentHTTP...but not sure we do... both designs are valid and have tradeoffs and Im on the fence on this.

Reply:
Im very torn on these different design decisions, not strong feelings so far. If we stick with tracing I think tracing.InstrumentHTTP could be enough since it should have only a single tracing related function (but I specified the procotol because we could have other forms of instrumentation that are tracing related but are not HTTP...maybe actual tracing =P).

Regarding the OrgID, I would also place it on the tracing.InstrumentHTTP..I just want something that solves all our problems regarding basic traced data on HTTP..but maybe that is bad...and in that case we would need a different package like organization.InstrumentHTTP...but not sure we do... both designs are valid and have tradeoffs and Im on the fence on this.